### DISCUSSION TITLE
Escaping single quote character in vue

### DISCUSSION THREAD

**Discussion by Masaomix** [zauberzeug/nicegui#2790] [Q&A] [answered]
### Question

Hi,

with the help of Falko (https://github.com/zauberzeug/nicegui/discussions/2721#discussioncomment-8838709), i managed to pass a variable to a vue component. Here is the simplified code:

```py
from nicegui import ui
columns = [
    {'name': 'name', 'label': 'Name', 'field': 'name'},
    {'name': 'age', 'label': 'Age', 'field': 'age'},
]
rows = [
    {'name': 'Alice', 'age': 18},
    {'name': 'Bob', 'age': 21},
    {'name': 'Carol', 'age': 42},
]

search_name = 'Bob'

table = ui.table(columns=columns, rows=rows, row_key='name')

table.add_slot('body-cell-name', '''
    <q-td key="name" :props="props">
        <q-badge :color="props.value === \'''' + search_name + '''\' ? 'red' : 'green'">
            {{ props.value }}
        </q-badge>
    </q-td>
''')
ui.run(show= False, reload= False)
```

this works very nice for the given table. But if the name in a row contains a single quote character , it breaks. 

for example:
```py
 {'name': "Bob's", 'age': 21},
search_name = "Bob's"
```

what do i have to change to make this work, regardless if the string contained in the row contains a ' - or not?
In my program i work with a very large table and it would possibly break things if i manipulate the row data directly. 

Thank's in advance.

**Comment by falkoschindler** [comment, marked as answer]
Hi @Masaomix,

This is an interesting puzzle! I thought about it over the weekend but just couldn't find a way to embed the search string such that it works for single and double quotes as well as backticks. I think we need to filter on the server and use something like the row index to communicate the result to the client:

```py
columns = [
    {'name': 'name', 'label': 'Name', 'field': 'name'},
    {'name': 'age', 'label': 'Age', 'field': 'age'},
]
rows = [
    {'name': 'Alice', 'age': 18},
    {'name': "Bob's", 'age': 21},
    {'name': 'Carol', 'age': 42},
]

search_name = "Bob's"
search_index = next((i for i, row in enumerate(rows) if row['name'] == search_name), None)

table = ui.table(columns=columns, rows=rows, row_key='name')

table.add_slot('body-cell-name', '''
    <q-td key="name" :props="props">
        <q-badge :color="props.rowIndex === ''' + str(search_index) + ''' ? 'red' : 'green'">
            {{ props.value }}
        </q-badge>
    </q-td>
''')
```

**Comment by Masaomix** [comment]
Thank you again! This problem was driving me crazy. In the meantime i made a workaround - but your solution is way better.
